src/main/java/com/google/devtools/build/lib/bazel/repository/DecompressorDescriptor.java

```diff
@@ -33,17 +34,21 @@ public class DecompressorDescriptor {
   private final Path repositoryPath;
   private final Optional<String> prefix;
   private final boolean executable;
+  private final Map<String, String> renameFiles;
   private final Decompressor decompressor;
 
   private DecompressorDescriptor(
       String targetKind, String targetName, Path archivePath, Path repositoryPath,
-      @Nullable String prefix, boolean executable, Decompressor decompressor) {
+      @Nullable String prefix, boolean executable,
+      @Nullable Map<String, String> renameFiles,
```

Comment:
this doesn't need to be nullable (it's never null anyway, and null has the same semantics as empty). also remove all the null checks